tests/gtests/graph/unit/utils/test_pattern_matcher_cpu.cpp

```diff
@@ -1108,6 +1108,46 @@ TEST(test_utils_pattern_matcher, OptionalWithLargerPort) {
     ASSERT_EQ(fusion_ops.size(), 3U);
 }
 
+TEST(test_utils_pattern_matcher, CommutativeSelect) {
+    auto graphp = std::make_shared<pb_graph_t>();
+    // Pattern that captures
+    // MatMul  Add
+    //   \     /
+    //    Select
+    auto pmatmul = graphp->append_op(MatMul);
```

Comment:
Please add another input for select's 0-th input (e.g. GreaterEqual), we want to know if the three inputs (both commutative and non-commutative) can all be matched

Reply:
0-th input for select has been added.

Reply:
The logic of `match_commutative_inputs()` has been modified: when `node_input_offset` and `op_input_offset` are not equal, it will check whether both belong to `commutative_inputs_`. If they do not, `match_input_by_offset()` will not be called.